tests/test_java.py

```diff
@@ -718,3 +720,284 @@ def test_java_frames_include_no_semicolons(
+                    "type": "size_t",
+                    "value": "1048576",
+                    "origin": "ergonomic",
+                    "kind": ["product"],
+                },
+                {
+                    "name": "GCDrainStackTargetSize",
+                    "type": "uintx",
+                    "value": "64",
+                    "origin": "ergonomic",
+                    "kind": ["product"],
+                },
+                {
+                    "name": "InitialHeapSize",
+                    "type": "size_t",
+                    "value": "264241152",
+                    "origin": "ergonomic",
+                    "kind": ["product"],
+                },
+                {
+                    "name": "MarkStackSize",
+                    "type": "size_t",
+                    "value": "4194304",
+                    "origin": "ergonomic",
+                    "kind": ["product"],
+                },
+                {
+                    "name": "MaxHeapSize",
+                    "type": "size_t",
+                    "value": "4196401152",
+                    "origin": "ergonomic",
+                    "kind": ["product"],
+                },
+                {
+                    "name": "MaxNewSize",
+                    "type": "size_t",
+                    "value": "2517630976",
+                    "origin": "ergonomic",
+                    "kind": ["product"],
+                },
+                {
+                    "name": "MinHeapDeltaBytes",
+                    "type": "size_t",
+                    "value": "1048576",
+                    "origin": "ergonomic",
+                    "kind": ["product"],
+                },
+                {
+                    "name": "SegmentedCodeCache",
+                    "type": "bool",
+                    "value": "true",
+                    "origin": "ergonomic",
+                    "kind": ["product"],
+                },
+                {
+                    "name": "UseCompressedClassPointers",
+                    "type": "bool",
+                    "value": "true",
+                    "origin": "ergonomic",
+                    "kind": ["lp64_product"],
+                },
+                {
+                    "name": "UseCompressedOops",
+                    "type": "bool",
+                    "value": "true",
+                    "origin": "ergonomic",
+                    "kind": ["lp64_product"],
+                },
+                {"name": "UseG1GC", "type": "bool", "value": "true", "origin": "ergonomic", "kind": ["product"]},
+            ],
+        ),
+    ],
+)
+def test_get_default_jvm_flags(
+    tmp_path: Path,
+    application_pid: int,
+    assert_collapsed: AssertInCollapsed,
+    expected_flags: List[Dict[str, Union[str, List[str]]]],
+) -> None:
+    with make_java_profiler(
+        frequency=99,
+        storage_dir=str(tmp_path),
+        java_async_profiler_mode="cpu",
```

Comment:
No reason to override parameters to `make_java_profiler`, you don't profile with it anyway